src/StudioComponents/Slider.lua

```diff
@@ -105,76 +85,92 @@ return function(props: SliderProperties): TextButton
 		return (unwrap(props.ZIndex) or 0) + 1
 	end)
 
-	local handleFill = themeProvider:GetColor(Enum.StudioStyleGuideColor.Button, handleModifier)
-	local handleBorder = themeProvider:GetColor(Enum.StudioStyleGuideColor.Border, handleModifier)
-
-	local newSlider = BoxBorder {
-		[Children] = New "TextButton" {
-			Name = "Slider",
-			Text = "",
-			Active = false,
-			AutoButtonColor = false,
-			Size = UDim2.new(1, 0, 0, 22),
-			ZIndex = zIndex,
-			BorderSizePixel = 0,
-			BackgroundColor3 = Spring(themeProvider:GetColor(Enum.StudioStyleGuideColor.InputFieldBackground, mainModifier), 40),
-			[Cleanup] = cleanupInputValueObserver,
-
-			[Children] = {
-				New "Frame" {
+	local mainModifier = Computed(function()
+		local isDisabled = not unwrap(isEnabled)
+		if isDisabled then
+			return Enum.StudioStyleGuideModifier.Disabled
+		end
+		return Enum.StudioStyleGuideModifier.Default
+	end)
+
+	local handleModifier = Computed(function()
+		local isDisabled = not unwrap(isEnabled)
+		local isHovering = unwrap(isHovering)
+		local isDragging = unwrap(isDragging)
+		if isDisabled then
+			return Enum.StudioStyleGuideModifier.Disabled
+		elseif isDragging then
+			return Enum.StudioStyleGuideModifier.Selected
+		elseif isHovering then
+			return Enum.StudioStyleGuideModifier.Hover
+		end
+		return Enum.StudioStyleGuideModifier.Default
+	end)
+
+	local handleFill = themeProvider:GetColor(Enum.StudioStyleGuideColor.Button)
+	local handleBorder = themeProvider:GetColor(Enum.StudioStyleGuideColor.InputFieldBorder, handleModifier)
+
+	local newSlider = New "Frame" {
+		Name = "Slider",
+		Size = UDim2.new(1, 0, 0, 22),
+		ZIndex = zIndex,
+		BackgroundTransparency = 1,
+		[Cleanup] = cleanupInputValueObserver,
+
+		[Children] = {
+			BoxBorder {
+				Color = themeProvider:GetColor(Enum.StudioStyleGuideColor.InputFieldBorder),
+
+				[Children] = New "Frame" {
```

Comment:
I don't see a color provided this with this bar frame. Is this frame the correct color in light theme?

Reply:
What do you mean? Color is line 122 and line 134.